src/classes/ALLO_PaymentSync_TEST.cls

```diff
@@ -618,6 +618,213 @@ private with sharing  class ALLO_PaymentSync_TEST {
             'Inserting or updating opportunity and payment allocations for the same opportunity failed to cause the expected exception.');
     }
 
+    /*******************************************************************************************************
+    * @description Test when an opportunity is overscheduled, has payment level allocations changed, 
+    * then opportunity level allocations changed. The payment level allocations should go back to the opportunity
+    * level allocation. 
+    ********************************************************************************************************/
+    @isTest
+    private static void testAllocationAmountIncreasedOnOpportunityOverscheduled() {
+        setupForTests();
+        makePayments(5000, 10);
+        makeGAUs(2);
+        Id GAUId = gaus[1].Id;
+        List<Allocation__c> alloForInsert = new List<Allocation__c>();
+
+        alloForInsert.add(new Allocation__c(Payment__c=payments[0].Id, Percent__c = 50, General_Accounting_Unit__c = GAUId));
+
+        insert alloForInsert;
+
+        Allocation__c oppAllocation = [select Id,
+                                Opportunity__c,
+                                Payment__c,
+                                General_Accounting_Unit__c,
+                                General_Accounting_Unit__r.Name,
+                                Amount__c,
+                                Percent__c
+                       from Allocation__c
+                       where Opportunity__c = :opportunity.Id
+                       AND General_Accounting_Unit__c =: GAUId
+                       order by Amount__c DESC];
+
+        System.assert(oppAllocation.Percent__c == 5, 'This is the opportunity allocation amount: ' + oppAllocation.percent__c);
+
+        oppAllocation.Percent__c = 50;
+        Test.startTest();
+        update oppAllocation;
+
+        Allocation__c paymentAllocation = [select Id,
+                                Opportunity__c,
+                                Payment__c,
+                                General_Accounting_Unit__c,
+                                General_Accounting_Unit__r.Name,
+                                Amount__c,
+                                Percent__c
+                       from Allocation__c
+                       where Payment__c = :payments[0].Id
+                       AND General_Accounting_Unit__c =: GAUId
+                       order by Amount__c DESC];
+
+        System.assert(paymentAllocation.Percent__c == 50, 'The allocation percent is incorrect. It should be 50, but it is actually: ' + paymentAllocation);
+        Test.stopTest();
+    }
+
+    /*******************************************************************************************************
+    * @description Test when an opportunity is underscheduled, has payment level allocations changed, 
```

Comment:
Can you be a little more specific on each test's description, meaning enough that we know the difference between each of the tests.  The three tests you added, have the same description (even the overscheduled one).